tests/test_authority.py

```diff
@@ -123,3 +128,104 @@ class MockResponse(object):
+        app.get_accounts()  # This could make an instance metadata call for authority aliases
+        instance_metadata.assert_not_called()
+
+    def test_by_default_adfs_should_skip_validation_and_instance_metadata(
+            self, instance_metadata, known_to_microsoft_validation, _):
+        """Not strictly required, but when/if we already supported it, we better keep it"""
+        app = msal.PublicClientApplication("id", authority="https://contoso.com/adfs")
+        known_to_microsoft_validation.assert_not_called()
+        app.get_accounts()  # This could make an instance metadata call for authority aliases
+        instance_metadata.assert_not_called()
+
+    def test_by_default_b2c_should_skip_validation_and_instance_metadata(
+            self, instance_metadata, known_to_microsoft_validation, _):
+        """Not strictly required, but when/if we already supported it, we better keep it"""
+        app = msal.PublicClientApplication(
+            "id", authority="https://login.b2clogin.com/contoso/b2c_policy")
+        known_to_microsoft_validation.assert_not_called()
+        app.get_accounts()  # This could make an instance metadata call for authority aliases
+        instance_metadata.assert_not_called()
+
+@patch("msal.authority.tenant_discovery", return_value=_mocked_tenant_discovery_response)
+@patch("msal.authority.instance_discovery")
+@patch.object(msal.PublicClientApplication, "_get_instance_metadata", return_value=[])
+class TestMsalAuthorityBehaviorsWithKnownToDeveloperAuthorities(unittest.TestCase):
+
+    def test_known_authorities_should_allow_multiple_configuration_but_only_one_modification(self, *mocks):
+        msal.PublicClientApplication.set_known_authority_hosts(["private.cloud"])
+        msal.PublicClientApplication.set_known_authority_hosts(["private.cloud"])
+        with self.assertRaises(ValueError):
+            msal.PublicClientApplication.set_known_authority_hosts(["hacked.com"])
+
+    def test_validate_authority_boolean_should_not_coexist_with_known_authorities(self, *mocks):
+        msal.PublicClientApplication.set_known_authority_hosts(["private.cloud"])
+        with self.assertRaises(ValueError):
+            msal.PublicClientApplication(
+                "id", authority="https://unknown.com/common", validate_authority=False)
+
+    def test_known_to_developer_authority_should_skip_validation_and_instance_metadata(
+            self, instance_metadata, known_to_microsoft_validation, _):
+        msal.PublicClientApplication.set_known_authority_hosts(["private.cloud"])
+        app = msal.PublicClientApplication("foo", authority="https://private.cloud/foo")
+        known_to_microsoft_validation.assert_not_called()
+        app.get_accounts()  # This could make an instance metadata call for authority aliases
+        instance_metadata.assert_not_called()
+
+    def test_a_known_to_microsoft_authority_should_skip_validation_but_still_use_instance_metadata(
```

Comment:
nit 2 test cases seem to have the same name. It might be working in Python but does not work in Java. For the sake of consistency, can this be renamed to something else?

Reply:
Oh, they belong to 2 different classes. Each class is conceptually a group of test scenarios. Group 1 starting from line 140 is testing the absence of new `known_authority_hosts`, the group 2 starting from line 179 is for with the new setting. This group separation makes same name method work.

I could have also mention that "with or without known hosts" into the test case name, but as you can see, each test case's name is already around full-line long. :-)